hydromt/models/model_mesh.py

```diff
@@ -253,33 +293,163 @@ def set_mesh(
                     f"Cannot set mesh from {str(type(data).__name__)} without a name."
                 )
             data = data.to_dataset()
+
+        # Checks on grid topology
+        # TODO: check if we support setting multiple grids at once. For now just one
+        if len(data.ugrid.grids) > 1:
+            raise ValueError(
+                "set_mesh methods only supports adding data to one grid at a time."
```

Comment:
When setting a mesh, this convertion will lose the optional attributes (accoding to xugrid and ugrid convention) that are *not* optional for plugin.
```suggestion
            data = data.to_dataset(optional_attributes = True)
```

Reply:
I guess you did some commits after the review so the line tracking from git is not working anymore (change is applied to another line...). But I will change and apply your suggestion in a local commit :)